react/features/notifications/middleware.js

```diff
@@ -11,26 +11,99 @@ import {
+import {
+    CLEAR_NOTIFICATIONS,
+    HIDE_NOTIFICATION,
+    SHOW_NOTIFICATION
+} from '../notifications/actionTypes';
 import { PARTICIPANTS_PANE_OPEN } from '../participants-pane/actionTypes';
 
 import {
     clearNotifications,
-    hideRaiseHandNotifications,
+    hideNotification,
     showNotification,
     showParticipantJoinedNotification,
     showParticipantLeftNotification
 } from './actions';
-import { NOTIFICATION_TIMEOUT_TYPE } from './constants';
-import { joinLeaveNotificationsDisabled } from './functions';
+import {
+    NOTIFICATION_TIMEOUT_TYPE,
+    RAISE_HAND_NOTIFICATION_ID
+} from './constants';
+import { areThereNotifications, joinLeaveNotificationsDisabled } from './functions';
 
 /**
  * Middleware that captures actions to display notifications.
  *
  * @param {Store} store - The redux store.
  * @returns {Function}
  */
+
+const timers = new Map();
+
+const createTimeoutId = (notification, dispatch) => {
+    const {
+        timeout,
+        uid
+    } = notification;
+    const timerID = setTimeout(() => {
+        dispatch(hideNotification(uid));
+    }, timeout);
+
+    timers.set(uid, timerID);
+};
+
 MiddlewareRegistry.register(store => next => action => {
+
     switch (action.type) {
+    case CLEAR_NOTIFICATIONS: {
+        if (navigator.product !== 'ReactNative') {
+            const { getState } = store;
+            const state = getState();
+            const _visible = areThereNotifications(state);
+            const { notifications } = state['features/notifications'];
+            const _notifications = _visible ? notifications : [];
+
+            for (const notification of _notifications) {
+                if (timers.has(notification.uid)) {
+                    const timeout = timers.get(notification.uid);
+
+                    clearTimeout(timeout);
+                    timers.delete(notification.uid);
+                }
+            }
+            timers.clear();
+        }
+        break;
+    }
+    case SHOW_NOTIFICATION: {
+        if (navigator.product !== 'ReactNative') {
+            const { dispatch, getState } = store;
+            const state = getState();
+            const _visible = areThereNotifications(state);
+            const { notifications } = state['features/notifications'];
+            const _notifications = _visible ? notifications : [];
+
+            for (const notification of _notifications) {
+                if (!timers.has(notification.uid) && notification.timeout) {
+                    createTimeoutId(notification, dispatch);
+                } else {
+
+                    const timer = timers.get(notification.uid);
+
+                    clearTimeout(timer);
+                    timers.delete(notification.uid);
+                    createTimeoutId(notification, dispatch);
+                }
+            }
+        }
+        break;
+    }
+    case HIDE_NOTIFICATION: {
+        if (navigator.product !== 'ReactNative') {
+            timers.clear();
```

Comment:
This is incorrect, this is about hiding one notification, you should just remove the timer for it, not all of them.